nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,386 @@
+    return colors, vmin, vmax, our_cmap, norm, abs_threshold
+
+
+def _encode(a):
+    try:
+        data = a.tobytes()
+    except AttributeError:
+        # np < 1.9
+        data = a.tostring()
+    return base64.b64encode(data).decode('utf-8')
+
+
+def _decode(b, dtype):
+    return np.frombuffer(base64.b64decode(b.encode('utf-8')), dtype)
+
+
+def to_plotly(mesh):
+    mesh = surface.load_surf_mesh(mesh)
+    x, y, z = map(_encode, np.asarray(mesh[0].T, dtype='<f4'))
+    i, j, k = map(_encode, np.asarray(mesh[1].T, dtype='<i4'))
+    info = {
+        "_x": x,
+        "_y": y,
+        "_z": z,
+        "_i": i,
+        "_j": j,
+        "_k": k,
+    }
+    return info
+
+
+def _to_color_strings(colors):
+    colors = np.asarray(colors)
+    colors = np.asarray(colors * 255, dtype='uint8')
+    colors = ['#{:02x}{:02x}{:02x}'.format(*row) for row in colors]
+    return colors
+
+
+def _get_vertexcolor(surf_map, cmap, norm,
+                     absolute_threshold=None, bg_map=None):
+    vertexcolor = cmap(norm(surf_map).data)
+    if absolute_threshold is None:
+        return _to_color_strings(vertexcolor)
+    if bg_map is None:
+        bg_map = np.ones(len(surf_map)) * .5
+        bg_vmin, bg_vmax = 0, 1
+    else:
+        bg_map = surface.load_surf_data(bg_map)
+        bg_vmin, bg_vmax = np.min(bg_map), np.max(bg_map)
+    bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
+    bg_color = mpl.cm.get_cmap('Greys')(bg_norm(bg_map))
+    vertexcolor[np.abs(surf_map) < absolute_threshold] = bg_color[
+        np.abs(surf_map) < absolute_threshold]
+    return _to_color_strings(vertexcolor)
+
+
+def one_mesh_info(surf_map, surf_mesh, threshold=None, cmap=cm.cold_hot,
+                  black_bg=False, bg_map=None, symmetric_cmap=True):
+    """Prepare info for plotting one surface map on a single mesh."""
+    info = {}
+    colors, cmin, cmax, cmap, norm, abs_threshold = colorscale(
+        cmap, surf_map, threshold, symmetric_cmap=symmetric_cmap)
+    info['inflated_left'] = to_plotly(surf_mesh)
+    info['vertexcolor_left'] = _get_vertexcolor(
+        surf_map, cmap, norm, abs_threshold, bg_map)
+    info["cmin"], info["cmax"] = float(cmin), float(cmax)
+    info['black_bg'] = black_bg
+    info['full_brain_mesh'] = False
+    info['colorscale'] = colors
+    return info
+
+
+def _check_mesh(mesh):
+    if not isinstance(mesh, str):
+        assert isinstance(mesh, collections.Mapping)
+        assert {'pial_left', 'pial_right', 'sulc_left', 'sulc_right',
+                'infl_left', 'infl_right'}.issubset(mesh.keys())
+        return mesh
+    mesh = datasets.fetch_surf_fsaverage(mesh)
+    return mesh
+
+
+def full_brain_info(volume_img, mesh='fsaverage5', threshold=None,
+                    cmap=cm.cold_hot, black_bg=False, symmetric_cmap=True,
+                    vol_to_surf_kwargs={}):
+    """Project 3d map on cortex; prepare info to plot both hemispheres."""
```

Comment:
Could you add more details here ? What type of info are you preparing here ?

If you have time, I think doc strings are super useful here.